### ISSUE TITLE
Create scaffold-experiment skill for automated experiment setup from experiment designs

### ISSUE THREAD

**Issue by niznik-dev** [niznik-dev/predicting-zygosity#154]
## Summary

Create a `scaffold-runs` skill that takes the output from `design-experiment` and automatically generates the directory structure, configuration files, and SLURM scripts for all runs in an experiment.

## Motivation

Currently, after running `design-experiment`, users must manually:
1. Create directories for each run
2. Write `setup_finetune.yaml` for each run with specific parameters
3. Run `setup_finetune.py` for each run to generate configs
4. Submit each job individually

For experiments with many runs (e.g., 8 runs in a hyperparameter sweep), this is tedious and error-prone. The `scaffold-runs` skill automates this entire workflow.

## Input

The skill should read the `experiment_summary.md` file generated by `design-experiment`, which contains:
- List of all runs with their parameters (LoRA rank, learning rate, etc.)
- Model paths and dataset paths
- Common configuration (batch size, epochs, GPUs, etc.)
- Experiment directory path

Example: `/scratch/gpfs/MSALGANIK/niznik/cap_4L_lora_lr_sweep_2025-10-22/experiment_summary.md`

## Expected Behavior

For each run listed in the experiment summary's "All Runs" table:

1. **Create run directory**
   - Pattern: `{experiment_dir}/{params_that_vary}/`
   - Example: `/scratch/gpfs/MSALGANIK/niznik/cap_4L_lora_lr_sweep_2025-10-22/rank8_lr1e-5/`
   - Only include parameters that vary across runs in the directory name
   - Constants (like model name) should not be in the directory name

2. **Generate `setup_finetune.yaml`**
   - Use appropriate template from `tasks/capitalization/templates/finetuning/`
   - Populate with run-specific parameters:
     - `lora_rank`: from run table
     - `learning_rate`: from run table
     - `my_wandb_run_name`: derived from run name or directory name
     - `batch_size`, `epochs`, `num_gpus`: from common config
     - `model_checkpoint`, `dataset`: from resources section
   - Use values from `claude.local.md` for environment-specific settings (conda env, output dirs, etc.)

3. **Run setup script**
   - Execute: `python ../../tools/torchtune/setup_finetune.py`
   - This generates `finetune.yaml` and `finetune.slurm`

4. **Verify outputs**
   - Confirm all expected files exist
   - Report any errors or missing files

## Output

After running, the experiment directory should look like:

```
cap_4L_lora_lr_sweep_2025-10-22/
├── experiment_summary.md
├── design-experiment.log
├── rank8_lr1e-5/
│   ├── setup_finetune.yaml
│   ├── finetune.yaml
│   └── finetune.slurm
├── rank8_lr5e-5/
│   ├── setup_finetune.yaml
│   ├── finetune.yaml
│   └── finetune.slurm
├── rank16_lr1e-5/
│   ├── setup_finetune.yaml
│   ├── finetune.yaml
│   └── finetune.slurm
... (remaining runs)
```

The skill should also update the experiment summary's status section or create a log indicating which runs were successfully scaffolded.

## Usage Example

```bash
# After running design-experiment
cd /scratch/gpfs/MSALGANIK/niznik/cap_4L_lora_lr_sweep_2025-10-22
claude scaffold-runs
```

Or via Claude Code chat:
```
User: Please scaffold the runs for this experiment
Claude: *runs scaffold-runs skill*
```

## Technical Considerations

1. **Directory naming**: Must intelligently extract only varying parameters for directory names
2. **Template selection**: Skill should determine which template to use based on dataset format in experiment summary
3. **Path resolution**: Must handle paths correctly whether running from experiment dir or elsewhere
4. **Error handling**: Should gracefully handle missing models, datasets, or invalid configs
5. **Idempotency**: Should be safe to re-run (e.g., if one run failed to generate properly)
6. **Validation**: Should verify model and dataset paths exist before generating configs
7. **Environment config**: Should read from `claude.local.md` for user-specific settings

## Scope

**In scope for initial version:**
- Fine-tuning runs only (not evaluation)
- Capitalization task (can generalize later)
- Reading experiment_summary.md format
- Generating directories + configs for all runs
- Intelligent directory naming based on varying parameters

**Out of scope (future enhancements):**
- Automatic job submission (separate skill or flag)
- Evaluation run setup (separate skill)
- Supporting other tasks beyond capitalization
- Dependency management between runs

## Related

- This skill complements `design-experiment` (issue #148)
- Previously considered as two separate skills: `generate-dirs` and `generate-slurm-scripts`
- Part of the larger workflow automation for cruijff_kit

## Success Criteria

- [ ] Skill can parse experiment_summary.md correctly
- [ ] Generates directories for all runs with names based only on varying parameters
- [ ] Creates valid setup_finetune.yaml for each run with correct parameters
- [ ] Successfully runs setup_finetune.py for each run
- [ ] All runs have finetune.yaml and finetune.slurm ready to submit
- [ ] Provides clear feedback about success/failure for each run
- [ ] Handles errors gracefully (missing files, invalid paths, etc.)

**Comment by niznik-dev** [comment]
## Naming Convention Update

Renamed the skill from `scaffold-runs` to `scaffold-experiment` to establish a consistent verb-experiment naming pattern across the experimental workflow:

- `design-experiment` ✓ (already exists)
- `scaffold-experiment` ✓ (renamed from scaffold-runs)
- `run-experiment` (planned)
- `analyze-experiment` (planned)

This naming makes it clearer that:
1. These skills are all part of the same experimental workflow
2. Each skill operates on an entire experiment, not individual runs
3. The verb-experiment pattern is easier to remember and discover

The skill functionality remains unchanged - it still scaffolds all runs in an experiment. Only the name has been updated for consistency.
